app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,50 @@
+class CommentsController < ApplicationController
+
+  def index
+    @comments = Comment.all
+  end
+
+  def new
+    @post = Post.find(params[:id])
```

Comment:
You'll need to use `:post_id` instead of `:id` to access the correct post. `:id` now corresponds with the comment's primary key -- not the post.